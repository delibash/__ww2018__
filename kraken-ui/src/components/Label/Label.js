import React from 'react';
import PropTypes from 'prop-types';
import {
  StyledLabel,
  StyledInfoLabel,
  StyledNodeLabel,
  StyledBlueLabel
} from './styled';

const Label = ({children, ...props}) => {
  return (
    <StyledLabel
      {...props.blueLabel && StyledBlueLabel}
      {...props.infoLabel && StyledInfoLabel}
      {...props.nodeLabel && StyledNodeLabel}
      {...props}
    >
      {children}
    </StyledLabel>
  );
};

Label.propTypes = {
  children: PropTypes.oneOfType(
    [
      PropTypes.string,
      PropTypes.node
    ]
  ),
  infoLabel: PropTypes.bool,
  nodeLabel: PropTypes.bool
};

export default Label;
