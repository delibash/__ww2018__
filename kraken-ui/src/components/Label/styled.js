import styled from 'styled-components';
import { colors } from './../../global/constants';

const StyledLabel = styled.span`
  padding: .6rem 1.2rem;
  font-size: ${props => props.fontSize || '1.3rem'}
  color: ${props => props.color || `${colors.BLUE}`};
  font-weight: ${props => props.fontWeight || 600 };
  border: ${props => props.border || `.1rem solid ${colors.GRAY}`};
  border-radius: ${props => props.borderRadius || '1.6rem'};
  background-color: ${props => props.backgroundColor || `${colors.LIGHTER_GRAY}`};
  display: ${props => props.display || ''};
  height: ${props => props.height || ''};
  width: ${props => props.width || ''}
  box-shadow: ${props => props.boxShadow || ''};
  align-items: ${props => props.alignItems || ''};
  position: ${props => props.position || ''};
`;

const StyledInfoLabel = ({
  borderRadius: '.2rem',
  color: `${colors.BLACK}`,
  fontWeight: 300,
  border: '0 none',
  backgroundColor: `${colors.LIGHTER_GRAY}`
});

const StyledBlueLabel = ({
  backgroundColor: `${colors.BLUE}`,
  color: `${colors.WHITE}`,
  border: '0 none'
});

const StyledNodeLabel = ({
  boxShadow: `${colors.BOX_SHADOW_ALL}`,
  borderRadius: '.2rem',
  border: '0 none',
  height: '4.2rem',
  fontWeight: 300,
  flex: 1,
  color: `${colors.BLACK}`,
  backgroundColor: `${colors.WHITE}`,
  display: 'flex',
  alignItems: 'center',
  width: '100%',
  position: 'relative'
});

export {
  StyledLabel,
  StyledInfoLabel,
  StyledNodeLabel,
  StyledBlueLabel
};