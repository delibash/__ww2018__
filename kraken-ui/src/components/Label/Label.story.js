import React from 'react';
import { storiesOf } from '@storybook/react';
import { colors } from './../../global/constants';
import Label from './Label';

storiesOf('Label', module)
  .add('Default Label', () => (
    <Label>Label</Label>
  ))
  .add('Square Label', () => (
    <Label
      borderRadius=".2rem"
      color={colors.BLACK}
      fontWeight="300"
      border="0 none"
      backgroundColor={colors.LIGHTER_GRAY}
    >Label</Label>
  ));


