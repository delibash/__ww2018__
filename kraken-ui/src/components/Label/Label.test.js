import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Label from './Label';

test('Default Label', () => {
  const tree = renderer.create(<Label>Label</Label>).toJSON();
  expect(tree).toMatchSnapshot();
});