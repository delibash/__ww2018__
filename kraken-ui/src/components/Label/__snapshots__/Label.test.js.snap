// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Label 1`] = `
.c0 {
  padding: .6rem 1.2rem;
  font-size: 1.3rem;
  color: rgba(100,107,255,1);
  font-weight: 600;
  border: .1rem solid rgba(0,0,0,.25);
  border-radius: 1.6rem;
  background-color: rgba(234,236,242,1);
}

<span
  className="c0"
>
  Label
</span>
`;
